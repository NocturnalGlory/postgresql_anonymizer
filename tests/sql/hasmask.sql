BEGIN;

CREATE EXTENSION IF NOT EXISTS anon CASCADE;

SELECT anon.start_dynamic_masking();

CREATE ROLE "WOPR";
SECURITY LABEL FOR anon ON ROLE "WOPR" IS 'MASKED';

CREATE ROLE hal LOGIN;
COMMENT ON ROLE hal IS 'MASKED';

CREATE ROLE jarvis;

-- FORCE update because COMMENT doesn't trigger the Event Trigger
SELECT anon.mask_update();

SELECT anon.hasmask('"WOPR"') IS TRUE;

SELECT anon.hasmask('hal') IS TRUE;

SELECT anon.hasmask('jarvis') IS FALSE;

SELECT anon.hasmask('postgres') IS FALSE;

SELECT anon.hasmask(NULL) IS FALSE;

-- Must return an error
SELECT anon.hasmask('does_not_exist');

ROLLBACK;

