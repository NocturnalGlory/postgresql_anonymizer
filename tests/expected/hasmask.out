BEGIN;
CREATE EXTENSION IF NOT EXISTS anon CASCADE;
SELECT anon.start_dynamic_masking();
 start_dynamic_masking 
-----------------------
 t
(1 row)

CREATE ROLE "WOPR";
SECURITY LABEL FOR anon ON ROLE "WOPR" IS 'MASKED';
CREATE ROLE hal LOGIN;
COMMENT ON ROLE hal IS 'MASKED';
CREATE ROLE jarvis;
-- FORCE update because COMMENT doesn't trigger the Event Trigger
SELECT anon.mask_update();
 mask_update 
-------------
 t
(1 row)

SELECT anon.hasmask('"WOPR"') IS TRUE;
 ?column? 
----------
 t
(1 row)

SELECT anon.hasmask('hal') IS TRUE;
 ?column? 
----------
 t
(1 row)

SELECT anon.hasmask('jarvis') IS FALSE;
 ?column? 
----------
 t
(1 row)

SELECT anon.hasmask('postgres') IS FALSE;
 ?column? 
----------
 t
(1 row)

SELECT anon.hasmask(NULL) IS FALSE;
 ?column? 
----------
 t
(1 row)

-- Must return an error
SELECT anon.hasmask('does_not_exist');
ERROR:  role "does_not_exist" does not exist
LINE 1: SELECT anon.hasmask('does_not_exist');
                            ^
ROLLBACK;
